flow/__main__.py

```diff
@@ -46,6 +50,46 @@ def main_init(args):
         )
 
 
+def main_template_create(args):
+    """Create custom template based on a specified or detected environment."""
+    if args.extends is None:
+        extend_template = environment.get_environment().template
+    else:
+        extend_template = args.extends[0]
+
+    project = get_project()
+    template_directory = project.fn(project._config.get("template_dir", "templates"))
+    os.makedirs(template_directory, exist_ok=True)
+
+    # grab and render custom template
+    jinja_env = jinja2.Environment(loader=jinja2.PackageLoader("flow"))
+    custom_template = jinja_env.get_template("custom_script_template.sh")
+    new_template = custom_template.render(extend_template=extend_template).strip(
+        string.whitespace
+    )
+
+    # If name is the default then it is a string if not it gets passed as a list
+    script_name = args.name if isinstance(args.name, str) else args.name[0]
+    script_path = os.path.join(template_directory, script_name)
+    try:
+        with open(script_path, "x") as fh:
+            fh.write(new_template)
+    except OSError as error:
+        if error.errno == errno.EEXIST:
+            logger.error(
+                f"Error while trying to create custom template. Delete '{script_path}' "
+                f"first and rerun command."
+            )
+        else:
+            logger.error(f"Error while trying to create custom template: '{error}'.")
+        raise
+    else:
+        print(
```

Comment:
It looks like all the other subparser commands print this kind of information to `sys.stderr`. I'm not sure if I like that, but we might want to be consistent with stdout or stderr among these types of messages.

https://github.com/glotzerlab/signac-flow/blob/5dc0a60f9b4c51f9a48dadc265b4a60a60373c4a/flow/template.py#L119

https://github.com/glotzerlab/signac-flow/blob/5dc0a60f9b4c51f9a48dadc265b4a60a60373c4a/flow/__main__.py#L36-L40

Reply:
I think this is a mistake, `stderr` should be used for errors. I would say we should remove output on success if we don't want this in `stdout`.

Reply:
Some tools print information to stderr that isn't an error. For example, signac-flow's progress bars are printed to stderr. Some suggest that stderr can be used for "diagnostic output": https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout

Basically, I don't agree that "`stderr` should [_only_] be used for errors." Either stdout or stderr is fine with me, so long as we are consistent about the choice with the other diagnostic messages linked in my first comment.

Reply:
The thread seems to be 6 one way and half a dozen the other with regard should outputting to stderr require an error exit code, and we fall on different sides of that. In this case, then if we want consistency, I can just switch to stderr.